### PR TITLE
incorporate changes probably from Samuel Kirschner and Yves

### PR DESCRIPTION
**Pull request by mathisdt**
- make matching pattern configurable via preference `extensions.replyasoriginalrecipient.patterns`
- use `manifest.json` instead of `install.rdf`
- compatibility to Thunderbird 68

also see the corresponding comments at https://blog.qiqitori.com/?p=194

### CODE REVIEW

File: chrome/content/composeOverlay.js [hunk 1]
```diff
@@ -25,11 +25,33 @@ var ReplyAsOriginalRecipient = {
...
-    /* Get original recipient */
-    originalHeader = this.getMessageHeaderFromURI(gMsgCompose.originalMsgURI);
-    originalRecipient = originalHeader.mime2DecodedRecipients;
-    if (originalRecipient.indexOf(",") != -1 || originalRecipient.indexOf("+") == -1)
-      return;
+    /* Get patterns preference (modified by Samuel Kirschner, according to the comment on https://blog.qiqitori.com/?p=194 ) */
+    var patterns = Components.classes["@mozilla.org/preferences-service;1"].getService(Components.interfaces.nsIPrefService).getBranch("extensions.replyasoriginalrecipient.").getCharPref("patterns"); // default is "*+*"
+    var mimeConvert = Components.classes["@mozilla.org/messenger/mimeconverter;1"]
+        .getService(Components.interfaces.nsIMimeConverter);
+	patterns = patterns.trim().replace(/[^a-zA-Z0-9 ]/g, '\\$&').replace(/\\\*/g, '.*').split(/ *\\, */).join('|');
```
Comment (qiqitori) [inline chrome/content/composeOverlay.js]: Hi, thank you for your patch. Can you add a comment for this line (and maybe break down the replaces/splits/joins into separate lines)? There are also some indentation issues that would be nicer to get fixed, but on the whole these commits are looking good.

**Comment by qiqitori** [comment]
BTW, can you briefly state in the commit message(s) which changes were needed to make this work in TB68?

File: chrome/content/composeOverlay.js (hunk 1 above)
Comment (mathisdt) [reply chrome/content/composeOverlay.js]: I did as you requested.

**Comment by mathisdt** [comment]
I'm afraid I can't tell what was needed to support TB68 - as I stated in the title, Samuel and Yves did the coding. I only created this PR to consolidate their changes which I got from the links in their comments on https://blog.qiqitori.com/?p=194.

**Comment by mathisdt** [comment]
@qiqitori, probably the use of `manifest.json` instead of `install.rdf` helps with the compatibility to TB68, but I'm not an extension developer so I don't really know if this is the only necessary change.

**Comment by mathisdt** [comment]
Can I help with something else, or are you confident enough with the changes to merge this PR?

**Comment by mathisdt** [comment]
@qiqitori I'd like to kindly ask if there is any obstacle to be overcome or if this PR can be merged? As far as I can see, all of your comments are addressed.

**Comment by qiqitori** [comment]
Indentation still looks a bit wonky but I'm gonna merge anyway

**Comment by qiqitori** [comment]
@mathisdt Merged. BTW -- I don't really remember how to test this extension (I just made it on request for someone). Can you confirm that the code still works and that the default behavior is still the same as the original extension?

**Comment by mathisdt** [comment]
First I used the original extension, then this version. Both worked similar, but with this version there are more options which makes it more fun to use. And yes, I can confirm that the default behaviour is the same (changing the sender address automatically for replies).

**Pull request merged**